bark/models/behavior/behavior_rss/behavior_rss.cpp

```diff
@@ -103,20 +114,36 @@ Trajectory BehaviorRSSConformant::Plan(
 }
 
 #ifdef RSS
-void BehaviorRSSConformant::ApplyRestrictionsToNominalModel(const ::ad::rss::state::AccelerationRestriction& acc_restrictions) {
-  bark::models::dynamic::AccelerationLimits acc_lim;
+AccelerationLimits BehaviorRSSConformant::ConvertRestrictions(
+    const ::ad::rss::state::AccelerationRestriction& acc_restrictions) {
+  AccelerationLimits acc_lim;
   acc_lim.lat_acc_left_max = acc_restrictions_.lateralLeftRange.maximum;
   acc_lim.lat_acc_right_max = acc_restrictions_.lateralRightRange.maximum;
   acc_lim.lon_acc_max = acc_restrictions_.longitudinalRange.maximum;
   acc_lim.lon_acc_min = acc_restrictions_.longitudinalRange.minimum;
   // TODO: Do we need minimum values as well?
   VLOG(4) << "RSS Response Acceleration Restrictions " << acc_restrictions_;
-  VLOG(4) << "AccelerationLimits for IDM " << acc_lim;
-  auto nominal_behavior =
-          std::dynamic_pointer_cast<BehaviorIDMLaneTracking>(
-            nominal_behavior_model_);
-  nominal_behavior->SetAccelerationLimits(acc_lim);
+  return acc_lim;
+}
+
+void BehaviorRSSConformant::ApplyRestrictionsToNominalModel(
+    const AccelerationLimits& limits) {
+  VLOG(4) << "AccelerationLimits for nominal model " << limits;
+  std::shared_ptr<BehaviorIDMLaneTracking> nominal_behavior_ptr =
+      std::dynamic_pointer_cast<BehaviorIDMLaneTracking>(
+          nominal_behavior_model_);
+  nominal_behavior_ptr->SetAccelerationLimits(limits);
 }
+
+void BehaviorRSSConformant::ApplyRestrictionsToSafetyModel(
+    const AccelerationLimits& limits) {
+  VLOG(4) << "AccelerationLimits for safety model " << limits;
+  std::shared_ptr<BehaviorIDMLaneTracking> safety_behavior_ptr =
+      std::dynamic_pointer_cast<BehaviorIDMLaneTracking>(
+          behavior_safety_model_);
+  safety_behavior_ptr->SetAccelerationLimits(limits);
+}
+
```

Comment:
I think it is ok to have ApplyRestrictionsToNominalModel and ApplyRestrictionsToSafetyModel. But it could be one template function